File: apps/patter-voice-sdk/libraries/typescript/src/tts/openai.ts
/** OpenAI TTS for Patter pipeline mode. */
import { OpenAITTS as _OpenAITTS } from "../providers/openai-tts";

/** Constructor options for the OpenAI `TTS` adapter. */
export interface OpenAITTSOptions {
  /** API key. Falls back to OPENAI_API_KEY env var when omitted. */
  apiKey?: string;
  voice?: string;
  model?: string;
  /** Voice-direction prompt (only honoured for gpt-4o-mini-tts and newer). */
  instructions?: string;
  /** Speech speed multiplier, must be in [0.25, 4.0] when set. */
  speed?: number;
  /**
   * Enable anti-aliasing LPF ahead of the 3:2 decimation. Defaults to
   * ``true`` (matches the provider default); set to ``false`` to opt out
   * for bit-exact downsample-only output.
   */
  antiAlias?: boolean;
}

/**
 * OpenAI TTS.
 *
 * @example
 * ```ts
 * import * as openai from "getpatter/tts/openai";
 * const tts = new openai.TTS();              // reads OPENAI_API_KEY
 * const tts = new openai.TTS({ apiKey: "sk-...", voice: "alloy" });
 * ```
 */
export class TTS extends _OpenAITTS {
  static readonly providerKey = "openai_tts";
  constructor(opts: OpenAITTSOptions = {}) {
    const key = opts.apiKey ?? process.env.OPENAI_API_KEY;
    if (!key) {
      throw new Error(
        "OpenAI TTS requires an apiKey. Pass { apiKey: 'sk-...' } or " +
          "set OPENAI_API_KEY in the environment.",
      );
    }
    super(
      key,
      opts.voice ?? "alloy",
      opts.model ?? "gpt-4o-mini-tts",
      opts.instructions ?? null,
      opts.speed ?? null,
      opts.antiAlias ?? true,
    );
  }
}

File: apps/patter-voice-sdk/libraries/python/getpatter/tts/openai.py
"""OpenAI TTS for Patter pipeline mode."""

from __future__ import annotations

import os
from typing import ClassVar

from getpatter.providers.openai_tts import OpenAITTS as _OpenAITTS

__all__ = ["TTS"]


class TTS(_OpenAITTS):
    """OpenAI streaming TTS.

    Example::

        from getpatter.tts import openai

        tts = openai.TTS()                  # reads OPENAI_API_KEY
        tts = openai.TTS(api_key="sk-...", voice="nova", model="tts-1-hd")
    """

    provider_key: ClassVar[str] = "openai_tts"

    def __init__(
        self,
        api_key: str | None = None,
        *,
        voice: str = "alloy",
        # gpt-4o-mini-tts: matches the underlying provider default AND the
        # TS facade — the same nominal config previously produced different
        # voice quality/latency/price per SDK (tts-1 here, 4o-mini there).
        model: str = "gpt-4o-mini-tts",
    ) -> None:
        key = api_key or os.environ.get("OPENAI_API_KEY")
        if not key:
            raise ValueError(
                "OpenAI TTS requires an api_key. Pass api_key='sk-...' or "
                "set OPENAI_API_KEY in the environment."
            )
        super().__init__(api_key=key, voice=voice, model=model)
